### PR TITLE
Nethermind Audit

### PR DESCRIPTION
**Pull request by hsouf**
Todo:
- [x] remove unused `l1_sender `on the  `handle_index_update` l1_handler

### CODE REVIEW

**Comment by hsouf** [comment]
`4.5.2` adds up unnecessary checks,  If for some reason the data passed was incorrect, a change of implementation will need to be done

**Comment by achab** [comment]
questions regarding the audit

code wise
- [ ] they mention several times "safety features" ("set and claim" or "set and confirm" or "timelock"). they want us to clarify of a change of proxy admin, for instance, can happen.
- [ ] 4.5.2: adds up unnecessary checks, if for some reason the data passed was incorrect, we can still can the implementation later
- [ ] 4.7.3. no need to check owner in approve_bridge because the guard already checks for the bridge address
- [ ] 4.7.4 should we add more checks on handle_deposit and handle_index_update
- [ ] we should probably upgrade to cairo 0.9.0. we'll give a try today 

documentation wise
- 1), 2) and 3) we will do: comments in the code (the cairo way)
- 4) should Aave integrate a better doc to its documentation or README is enough?
- 5) we have done that

File: README.md [hunk 1]
```diff
@@ -57,120 +47,95 @@ way yield from L1.
...
-## How does it work?
+## How it works
 
 ### Bridging aTokens from L1 to L2
 
+<a name="approve-bridge-tokens"></a>
 **Approve bridge tokens**
 
-L1 aTokens are approved on the bridge at `initiliaze` where `_approveBridgeTokens` is called internally to approve the provided array of aTokens in an array along with their corresponding static_a_tokens on L2. :warning: Gas limit concerns should apply here!
+Pairs of (L1 aToken, L2 static_a_token) are approved on the bridge in `initiliaze` function. The function `_approveBridgeTokens` is called internally to approve an array of aTokens with their corresponding static_a_tokens on L2.
```
Comment (hsouf) [inline README.md:92]: Why not simply write: L1 aTokens with their corresponding static_a_tokens on L2, are going to be approved on the l1 bridge in  `initialize` function.

Comment (hsouf) [inline README.md:92]: initiliaze -> initialize

File: README.md [hunk 2]
```diff
@@ -32,22 +31,13 @@ audited, might contain bugs and should not be used in production.
...
+Holding L1 aTokens lets you earn more tokens via two different mechanisms: (i) the amount of aTokens you hold increases over time and (ii) holding aTokens allows you to claim an accruing amount of Aave reward tokens. This bridge offers both mechanisms thanks to `static_a_tokens` on L2, and the equivalent of L1 Aave reward token on L2.
 
-The bridge is also shaped for liquidity providers who are able to assume the
-Ethereum gas cost of deposits and withdrawals as they transact large enough
-amounts. They will deposit on Aave Ethereum, bridge the `static_a_tokens` to
-Starknet and make them available for users there to buy and hold, accruing this
-way yield from L1.
+The bridge is also shaped for liquidity providers who are able to assume the Ethereum gas cost of deposits and withdrawals as they transact large enough amounts. They will deposit on Aave Ethereum, bridge the `static_a_tokens` to Starknet and make them available for users there to buy and hold, thus accruing yield from L1.
+
+We assume that L1 tokens approved by the bridge are pre-validated tokens, and that they are not deflationary.
```
Comment (hsouf) [inline README.md:40]: shouldn't we highlight this sentence?

File: README.md [hunk 3]
```diff
@@ -32,22 +31,13 @@ audited, might contain bugs and should not be used in production.
 
 ## Introduction
 
-For Aave, one of the main current and future goals is growth of liquidity and
-user base. As seen on side-chains with low transaction cost like Polygon or
-Avalanche, there is high demand to use the Aave protocol with small amounts to
-earn high yield. That's why we brought to you an initial phase of the Aave <>
-Starknet integration allowing deposit/withdrawal on Aave Ethereum by
-exclusively transacting on Starknet.
+This bridge project is the first step of Aave in Starknet ecosystem. The bridge allows users to deposit and withdraw their [aTokens](https://docs.aave.com/developers/tokens/atoken), and only `aTokens`, on StarkNet and get `static_a_tokens` instead. These `static_a_tokens` are equivalent to `aTokens`: the former grow in value when the latter grow in balance.
```
Comment (hsouf) [inline README.md:34]: This bridge project is the first step of Aave in Starknet ecosystem. The bridge allows users to deposit and withdraw their [aTokens](https://docs.aave.com/developers/tokens/atoken), and only `aTokens`, and receive wrapped tokens referred to as `static_a_tokens` on StarkNet. `static_a_tokens` are equivalent to `aTokens` except that the former grows in value when the latter grows in balance.

Comment (achab) [reply README.md:34]: Good suggestion, I'll adapt the sentence with your inputs

File: README.md (hunk 2 above)
Comment (achab) [reply README.md:40]: I think it's ok, most people won't care about this sentence. It's just an answer to Peckshield note.

File: contracts/l2/tokens/rewAAVE.cairo [hunk 4]
```diff
@@ -9,6 +9,13 @@ from starkware.cairo.common.bool import TRUE
 
 from contracts.l2.dependencies.openzeppelin.token.erc20.library import ERC20
 from contracts.l2.dependencies.openzeppelin.access.ownable import Ownable
+from contracts.l2.lib.version_initializable import VersionedInitializable
+
+#
+# Version
+#
+
+const REVISION_V1 = 1
```
Comment (hsouf) [inline contracts/l2/tokens/rewAAVE.cairo:19]: Let's use `REVISION` to rely on the value rather than the name.

**Review by achab — APPROVED** [review]
Ready to be merged

**Pull request merged**